### PR TITLE
CLI

### PR DESCRIPTION
**Pull request by iskandr**
Gives isovar a commandline interface and CSV output (https://github.com/hammerlab/isovar/issues/6)

### CODE REVIEW

**Comment by hammer** [comment]
I have some code review bandwidth and am interested in isovar. Does this PR represent the latest state of the project?

**Comment by iskandr** [comment]
Not quite, there are some fragments not checked in. I'll give you something
reviewable tomorrow though.

**Comment by hammer** [comment]
@iskandr ready for review?

**Comment by iskandr** [comment]
@hammer Not yet, still working through some edge cases I thought of last night. The parts that don't interact with read data (such as the `reference_context` module) are pretty stable though, if you want to start somewhere.

**Comment by iskandr** [comment]
@hammer `variant_reads`, `variant_sequences`, and `reads_at_locus` should also be more or less in their final shape, with passing unit tests. The only missing part now is that the translation logic is split across a few incomplete files, which I need to weave back together.

**Comment by iskandr** [comment]
@tavinathanson @ihodes @hammer If you all want to start reviewing, anything other than `protein_sequence` and its tests is fair game. 

I added some explanation of the design to the README but it's worth repeating a little here:

1) Reads are filtered from the BAM around each variant into [ReadAtLocus](https://github.com/hammerlab/isovar/blob/cli/isovar/reads_at_locus.py#L29) records. These contain the sequence of a read and the offsets into that sequence for the bases immediately before and after a variant. I found "anchoring" on reference bases adjacent to the variant was easier than trying to deal with the variant locus directly. 

2) Not every `ReadAtLocus` object contains the variant nucleotides, so these get further filtered down into [VariantRead](https://github.com/hammerlab/isovar/blob/cli/isovar/variant_reads.py#L37) objects. 

3) Multiple `VariantRead`s get collapsed into a [VariantSequence](https://github.com/hammerlab/isovar/blob/cli/isovar/variant_sequence.py#L33). These are the cDNA sequences we're going to try translating. 

4) Which reading frame to use? We're not really sure (without long reads + ribosomal footprint profiling), so instead we take a reasonable guess based on the reading frames of reference transcripts around this locus. I summarize the reference sequence at this locus and its associated reading frame in a [ReferenceContext](https://github.com/hammerlab/isovar/blob/cli/isovar/reference_context.py#L89) object. There's a weird little hierarchy of named tuples in that file, which I found simplified testing a lot (enabled me to right more tests and to think more clearly about little bits of functionality). 

5) Finally, the `VariantRead`s and `ReferenceContext`s of a variant get combined into zero or more [ProteinSequence](https://github.com/hammerlab/isovar/blob/cli/isovar/protein_sequence.py#L40) objects. Since we may only want the single best translation, both reference contexts and variant sequences are sorted first.

File: isovar/reads_at_locus.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    out if what's between those positions matches a variant happens later in
+    the `variant_reads` module.
+
+    Parameters
+    ----------
+    samfile : pysam.AlignmentFile
+
+    chromosome : str
+
+    base1_position_before_variant : int
```
Comment (tavinathanson) [inline isovar/reads_at_locus.py]: Just poking around; noticed that you have two `base1_position_before_variants`s vs. `after`.

**Comment by tavinathanson** [comment]
@iskandr I'm running into `AttributeError: 'module' object has no attribute 'cutils'` with `pysam 0.8.4`; doesn't complain with `0.9.0`.

**Comment by iskandr** [comment]
Fixes https://github.com/hammerlab/isovar/issues/12, https://github.com/hammerlab/isovar/issues/11, https://github.com/hammerlab/isovar/issues/10, https://github.com/hammerlab/isovar/issues/6. 

I need to more explicitly test for https://github.com/hammerlab/isovar/issues/4 since the B16 variants do have nearby germline variants but I haven't documented what they are.

**Pull request merged**